clients/src/main/java/org/apache/kafka/common/requests/BrokerRequestUtils.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.common.errors.InvalidRequestException;
+import org.apache.kafka.common.network.Send;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.Protocol;
+import org.apache.kafka.common.protocol.types.Struct;
+
+import java.nio.ByteBuffer;
+
+public class BrokerRequestUtils {
+
+    public static InboundRequest parseInboundRequest(ByteBuffer buffer, BrokerRequestContext context) {
+        try {
+            RequestHeader header = RequestHeader.parse(buffer);
+            return parseInboundRequest(header, buffer, context);
+        } catch (Throwable  ex) {
+            throw new InvalidRequestException("Error parsing request header. Our best guess of the apiKey is: " +
```

Comment:
Do we need this here since we have the catch in the other `parseInboundRequest`?